init/arch_specs/eessi_arch_arm.spec

```
@@ -7,3 +7,5 @@
 "aarch64/neoverse_n1"	"0x41"		"asimddp"		# AWS Graviton2
 "aarch64/neoverse_v1"	"ARM"		"asimddp svei8mm"
 "aarch64/neoverse_v1"	"0x41"		"asimddp svei8mm"	# AWS Graviton3
+"aarch64/google/axion"	"0x41"		"sve2 rng sm3 sm4 svesm4"	# Google Axion
+"aarch64/nvidia/grace"	"0x41"		"sve2 sm3 sm4 svesm4"		# NVIDIA Grace
```

@TopRichard My earlier suggestion was incorrect, since this leads to incorrect order on Axion systems, resulting in `nvidia/grace` being the "top" result while it should be `google/axion`.

Just flipping these is enough to make it correct, since that's more specific (also has `rng`):

```suggestion
"aarch64/nvidia/grace"	"0x41"		"sve2 sm3 sm4 svesm4"		# NVIDIA Grace
"aarch64/google/axion"	"0x41"		"sve2 rng sm3 sm4 svesm4"	# Google Axion
```

```
$ init/eessi_archdetect.sh -a cpupath
aarch64/google/axion:aarch64/nvidia/grace:aarch64/neoverse_v1:aarch64/neoverse_n1:aarch64/generic
```

So basically this means that binaries optimized for Grace can also be used on Axion, but will be "suboptimal" (since `rng` instructions will not be used).

To make sure we're not overlooking something, we should also add tests for `google/axion`, using the `/proc/cpuinfo` output available in https://github.com/EESSI/software-layer/issues/845#issuecomment-2827629221

---

Might be good to add a comment, why the order is important. There may be other ways to implement this, but they would require significantly more code.